browser/resources/settings/email_aliases_page/stories/index.tsx

```diff
@@ -0,0 +1,201 @@
+  emailAliasesClickOnSecureLogin: 'Click on the secure login link in the ' +
+    'email to access your account.',
+  emailAliasesDontSeeEmail: 'Don\'t see the email? Check your spam folder ' +
+    'or $1try again$2.',
+  emailAliasesBubbleDescription: 'Create a random email address that ' +
+    'forwards to your inbox while keeping your personal email private.',
+  emailAliasesBubbleLimitReached: 'You have reached the limit of 5 free ' +
+    'email aliases. Click "Manage" to re-use or delete an alias.',
+})
+
+export default {
+  title: 'Email Aliases',
+}
+
+const demoData = {
+  email: 'aguscr182@gmail.com',
+  aliases: [
+    {
+      email: 'horse.radish.record57@bravealias.com',
+      note: 'Alias for all my newsletters',
+    },
+    {
+      email: 'start.plane.division.laser42@bravealias.com',
+      domains: ['x.com'],
+      note: 'Twitter account'
+    },
+    {
+      email: 'racoon.pencil.test14@bravealias.com',
+      note: 'Marketplace email for Facebook'
+    }
+  ]
+} satisfies { email: string, aliases: Alias[] }
+
+class MockMappingService implements MappingService {
+  accountEmail: string
+  aliases: Map<string, Alias>
+  accountState: AccountState
+  accountRequestId: number
+  constructor(accountState: AccountState, accountEmail: string) {
+    this.accountState = accountState
+    this.accountEmail = accountEmail
+    this.aliases = new Map<string, Alias>();
+    for (const alias of demoData.aliases) {
+      this.aliases.set(alias.email, alias)
+    }
+  }
+
+  async createAlias (email: string, note: string) {
+    const alias = { email, note }
+    this.aliases.set(email, alias)
+  }
+
+  async getAliases () {
+    return [...this.aliases.values()]
+  }
+
+  async updateAlias (email: string, note: string, status: boolean) {
+    const alias = { email, note }
+    this.aliases.set(email, alias)
+  }
+
+  async deleteAlias (email: string) {
+    this.aliases.delete(email)
+  }
+
+  async generateAlias () {
+    let generated: string = ''
+    do {
+      generated = "mock-" + Math.random().toString().slice(2,6) +
+        "@bravealias.com"
+    } while (this.aliases.has(generated))
+    await new Promise(resolve => setTimeout(resolve, 1000))
+    return generated
+  }
+
+  async getAccountEmail () {
+    return this.accountEmail
+  }
+
+  async requestAccount (accountEmail: string) {
+    this.accountState = 'AwaitingAccount'
+    this.accountRequestId = window.setTimeout(() => {
+      this.accountEmail = accountEmail
+      this.accountState = 'AccountReady'
+    }, 5000);
+  }
+
+  async onAccountReady () {
+    while (this.accountState === 'AwaitingAccount') {
+      await new Promise(resolve => setTimeout(resolve, 250));
+    }
+    return this.accountState === 'AccountReady'
+  }
+
+  async cancelAccountRequest () {
+    this.accountState = 'NoAccount'
+    window.clearTimeout(this.accountRequestId)
+  }
+
+  async logout () {
+    this.accountState = 'NoAccount'
```

Comment:
shouldn't `NoAccount` be a constant?

Reply:
This is correct - we should avoid typescript enums as they're one of the few legacy Typescript features which generate code rather than compiling to nothing. Intellisense will suggest the correct values and we'll get a compile error for invalid values.

Reply:
Please do not resolve comments without a response from the person who originally made them

Reply:
but accountState is part of the mojo interface, right? We should not be passing strings for things like this between JS and cpp, they should be defined as constants/enums/etc... in the mojo file

Reply:
Right yeah, however in this case we don't have a mojom interface. Maybe its worth spinning up a separate PR with the mojo interface, as it'll make things a bit easier knowing the interface (and having all the types there).

Reply:
This has now been fixed with `AuthState`.